Makefile.dep

```diff
@@ -43,12 +43,16 @@ ifneq (,$(filter sntp,$(USEMODULE)))
   USEMODULE += xtimer
 endif
 
+ifneq (,$(filter netdev_layer_%,$(USEMODULE)))
+  USEMODULE += netdev_layer
+endif
+
 ifneq (,$(filter gnrc_netdev_default,$(USEMODULE)))
   USEMODULE += netdev_default
   USEMODULE += gnrc_netif
 endif
 
-ifneq (,$(filter netdev_ieee802154,$(USEMODULE)))
+ifneq (,$(filter netdev_layer_ieee802154,$(USEMODULE)))
```

Comment:
Independent from the separate type discussion @gebart started: I would argue that a rename of `netdev_ieee802154` and `netdev_eth` isn't really necessary.

Reply:
Yeah, I agree on that, I came to the same conclusion after I had to rename half of the ifdefs scattered through gnrc_netif and nib

Reply:
Reverted "new" netdev_layer_(eth|ieee802154) files and reworked netdev_(eth|ieee802154) instead.